main.go

```diff
@@ -0,0 +1,422 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"gator/internal/config"
+	"gator/internal/database"
+	"html"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	_ "github.com/lib/pq"
+)
+
+type Commands struct {
+	Handlers map[string]func(*config.State, config.Command) error
+}
+
+type RSSFeed struct {
+	Channel struct {
+		Title       string    `xml:"title"`
+		Link        string    `xml:"link"`
+		Description string    `xml:"description"`
+		Item        []RSSItem `xml:"item"`
+	} `xml:"channel"`
+}
+
+type RSSItem struct {
+	Title       string `xml:"title"`
+	Link        string `xml:"link"`
+	Description string `xml:"description"`
+	PubDate     string `xml:"pubDate"`
+}
+
+func handlerBrowse(state *config.State, cmd config.Command) error {
+	limit := 2
+	if len(cmd.Args) > 0 {
+		if parsedLimit, err := strconv.Atoi(cmd.Args[0]); err == nil {
+			limit = parsedLimit
+		} else {
+			return fmt.Errorf("Error: %s", cmd.Args[0])
+		}
+	}
+	posts, err := state.DB.GetPosts(context.Background(), int32(limit))
+	if err != nil {
+		return fmt.Errorf("Error: %w", err)
```

Comment:
неинформативный враппинг ошибок

```suggestion
		return fmt.Errorf("failed to get posts: %w", err)
```